tools/src/bin/run-fap.rs

```diff
@@ -62,17 +65,10 @@ impl From<io::Error> for Error {
     }
 }
 
-fn wait_for_idle(serial_cli: &mut serial::SerialCli) -> io::Result<()> {
-    loop {
-        serial_cli.send_and_wait_eol("loader info")?;
-        if serial_cli
-            .consume_response()?
-            .contains("No application is running")
-        {
-            break Ok(());
-        }
-        thread::sleep(Duration::from_millis(200));
-    }
+fn wait_for_idle(session: &mut RpcSession) -> io::Result<()> {
+    // TODO: need equivalent of "loader info" CLI command.
```

Comment:
There is probably an equivalent to this (to determine whether an app is still running), but I have not figured out what it is yet.